kafka/client.go

```diff
@@ -52,12 +56,13 @@ func NewClient(config *Config) (*Client, error) {
 		kafkaConfig: kc,
 	}
 
-	err = kc.Validate()
+	err = client.populateAPIVersions()
 	if err != nil {
 		return client, err
 	}
 
-	err = client.populateAPIVersions()
+	err = client.extractTopics()
```

Comment:
The client fetches metadata on initial connection. This extracts our topic map cache from it for reuse by others.